lib/dialects/mssql/AsyncQueue.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+class AsyncQueue {
+  constructor() {
+    this.previous = Promise.resolve();
+    this.closed = false;
+    this.rejectCurrent = () => {};
+  }
+  close() {
+    this.closed = true;
+    this.rejectCurrent(new Error('the connection was closed before this query could finish executing'));
```

Comment:
This should be a connection error, add a new error [here](https://github.com/sequelize/sequelize/tree/master/lib/errors/connection). Name suggestion `connection-queue-error`.

Reply:
would `connection-closed-error` be fine? What happens in other dialects when the connection is closed while a query is in flight?

Reply:
this is awkward: `ConnectionError` expects a `parent` error that was thrown by the dialect-specific connection, but in this case, we're throwing the error ourselves so there's no `parent` to wrap.  At the moment I made `ConnectionClosedError` inherit from `BaseError` instead

Reply:
In `connection-error.js`:

Reply:
In that case we can have an `async-queue-error`, define that error in this file. Use that as parent error in generic connection error thrown by dialect